swift/README.md

```diff
@@ -1,372 +1,1100 @@
+// Turn any generic type into a reference type using this Box class.
+final class Box<T> {
+  let value: T 
+  init(_ value: T) {
+    self.value = value
+  }
 }
 ```
 
-Place the block outside of the parenthesis when it is the last argument:
+Alternatively, consider using a `struct` instead since they do not allow inheritance by default.
+
+# Function Declarations
+
+Keep short function declarations on one line including the opening brace:
 
 ```swift
-// Calling UIView.animateWithDuration(duration: NSTimeInterval, animations: () -> Void)
+func reticulateSplines(spline: [Double]) -> Bool {
+  // reticulate code goes here
+}
+```
+
+For functions with long signatures, add line breaks after each parameter and indent them on the same level as the first parameter.
+
+```swift
+func reticulateSplines(spline: [Double], 
+                       adjustmentFactor: Double,
+                       translateConstant: Int,
+                       comment: String) -> Bool {
+  // reticulate code goes here
+}
+```
+
+## Closure Expressions
+
+Use trailing closure syntax only if there's a single closure expression parameter at the end of the argument list. Give the closure parameters descriptive names.
+
+**Preferred:**
+```swift
 UIView.animateWithDuration(1.0) {
-    label.alpha = 0.0 // Fade out the label
+  self.myView.alpha = 0
+}
+
+UIView.animateWithDuration(1.0,
+  animations: {
+    self.myView.alpha = 0
+  },
+  completion: { finished in
+    self.myView.removeFromSuperview()
+  }
+)
+```
+
+**Not Preferred:**
+```swift
+UIView.animateWithDuration(1.0, animations: {
+  self.myView.alpha = 0
+})
+
+UIView.animateWithDuration(1.0,
+  animations: {
+    self.myView.alpha = 0
+  }) { f in
+    self.myView.removeFromSuperview()
 }
 ```
 
-Place the closing `}` on the same line if the body of the block if empty:
+For single-expression closures where the context is clear, use implicit returns:
+
+```swift
+attendeeList.sort { a, b in
+  a > b
+}
+```
+
+Chained methods using trailing closures should be clear and easy to read in context. Decisions on spacing, line breaks, and when to use named versus anonymous arguments is left to the discretion of the author. Examples:
+
+```swift
+let value = numbers.map { $0 * 2 }.filter { $0 % 3 == 0 }.indexOf(90)
+
+let value = numbers
+   .map {$0 * 2}
+   .filter {$0 > 50}
+   .map {$0 + 10}
+```
+
+Place the closing curly brace (`}`) on the same line if the body of the block is empty:
 
 ```swift
 let callback: RpcMethodCallback = { result in }  // Do nothing
 ```
 
-Place bodies of a block on a new line
-=====================================
 
-No single line `if...else` statements
+## Types
+
+Always use Swift's native types when available. Swift offers bridging to Objective-C so you can still use the full set of methods as needed.
+
+**Preferred:**
+```swift
+let width = 120.0                                    // Double
+let widthString = (width as NSNumber).stringValue    // String
```

Comment:
I've changed the names of the variable -- can you think of a better example of bridging? I can't think of anything better than this 🤔